GitUI/Avatars/InitialsAvatarProvider.cs

```diff
@@ -3,42 +3,42 @@
 using System.Drawing.Drawing2D;
 using System.Drawing.Text;
 using System.Linq;
+using System.Threading.Tasks;
 using GitCommands;
+using JetBrains.Annotations;
 
 namespace GitUI.Avatars
 {
-    public class InitialsAvatarGenerator : IAvatarGenerator
+    /// <summary>
+    /// A provider that generates avatar images based on the initials of the user.
+    /// </summary>
+    public class InitialsAvatarProvider : IAvatarProvider
     {
         private int _unkownCounter = 0;
 
-        public Image GetAvatarImage(string email, string name, int imageSize)
-        {
-            if (AppSettings.GravatarFallbackAvatarType != GravatarFallbackAvatarType.AuthorInitials)
-            {
-                return null;
-            }
-
-            return GenerateAvatarImage(email, name, imageSize);
-        }
-
-        public Image GenerateAvatarImage(string email, string name, int imageSize)
+        /// <inheritdoc/>
+        public Task<Image> GetAvatarAsync([NotNull] string email, string name, int imageSize)
```

Comment:
Why not async?

Reply:
The interface is async, but the author initials are generated locally, are CPU bound, and wouldn't benefit from making it async because there is no awaitable dependency involved.